### ISSUE TITLE
fatal error in arch/hexagon/kernel/stacktrace.o

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#755]
```
$ ARCH=hexagon CROSS_COMPILE=hexagon-linux-gnu- make -j71 CC=clang LD=ld.lld defconfig
$ ARCH=hexagon CROSS_COMPILE=hexagon-linux-gnu- make -j71 CC=clang LD=ld.lld arch/hexagon/kernel/stacktrace.o
...
  CC      arch/hexagon/kernel/stacktrace.o
fatal error: error in backend: Invalid register name global variable
clang-10: error: clang frontend command failed with exit code 70 (use -v to see invocation)
clang version 10.0.0 (https://github.com/llvm/llvm-project.git 00bbe990c5d4472d5413479a539b3d6edbb3ca7a)
Target: hexagon-unknown-linux-gnu
Thread model: posix
InstalledDir: /android0/llvm-project/llvm/build/bin
clang-10: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
clang-10: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-10: note: diagnostic msg: /tmp/stacktrace-b0f4d4.c
clang-10: note: diagnostic msg: /tmp/stacktrace-b0f4d4.sh
clang-10: note: diagnostic msg: 

********************
make[1]: *** [scripts/Makefile.build:266: arch/hexagon/kernel/stacktrace.o] Error 70
make: *** [Makefile:1649: arch/hexagon/kernel] Error 2
```

**Comment by nickdesaulniers** [comment]
cc @androm3da

**Comment by nickdesaulniers** [comment]
I see one as well in net/core/dev.c but that may be an orthogonal issue.

**Comment by androm3da** [comment]
The fix applied in https://reviews.llvm.org/rL341658 added `r19`, looks like we just need to add `r30` for `stacktrace.c`.

FYI @SidManning

**Comment by nickdesaulniers** [comment]
Might as well add support for all registers.

**Comment by SidManning** [comment]
Probably better to update the kernel, try: fp = (unsigned long)__builtin_frame_address(0);

**Comment by nickdesaulniers** [comment]
Hows this look Sid, got an email addr I can attach to the patch to send upstream:

```diff
From ecfb297ad6810d06c55dfe4ed8abf4109bbd71da Mon Sep 17 00:00:00 2001
From: Nick Desaulniers <ndesaulniers@google.com>
Date: Fri, 25 Oct 2019 13:04:06 -0700
Subject: [PATCH] hexagon: work around compiler crash

Clang cannot translate the string "r30" into a valid register yet.

Link: https://github.com/ClangBuiltLinux/linux/issues/755
Suggested-by: Sid Manning <???>
Signed-off-by: Nick Desaulniers <ndesaulniers@google.com>
---
 arch/hexagon/kernel/stacktrace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/hexagon/kernel/stacktrace.c b/arch/hexagon/kernel/stacktrace.c
index 35f29423fda8..046cd2d26696 100644
--- a/arch/hexagon/kernel/stacktrace.c
+++ b/arch/hexagon/kernel/stacktrace.c
@@ -30,7 +30,7 @@ void save_stack_trace(struct stack_trace *trace)
 
 	low = (unsigned long)task_stack_page(current);
 	high = low + THREAD_SIZE;
-	fp = current_frame_pointer;
+	fp = (unsigned long)__builtin_frame_address(0);
 
 	while (fp >= low && fp <= (high - sizeof(*frame))) {
 		frame = (struct stackframe *)fp;
-- 
2.24.0.rc0.303.g954a862665-goog
```

**Comment by androm3da** [comment]
@SidManning is sidneym@quicinc.com

**Comment by nickdesaulniers** [comment]
https://lore.kernel.org/r/20191028155722.23419-1-ndesaulniers@google.com/

**Comment by nickdesaulniers** [comment]
https://llvm.org/pr43829

**Comment by SidManning** [comment]
This patch looks correct and should be fine to merge.
________________________________

**Comment by nickdesaulniers** [comment]
Krzystof fixed this in [99f51960fdb5559d6720281bff9a63041452bf9a](https://github.com/ClangBuiltLinux/llvm-project/commit/99f51960fdb5559d6720281bff9a63041452bf9a).

**Comment by nathanchance** [comment]
The kernel side work around was merged into mainline: https://git.kernel.org/linus/63e80314ab7cf4783526d2e44ee57a90514911c9
